spec/unit/settings_spec.rb

```diff
@@ -221,6 +221,27 @@ def metadata(setting)
       @settings.define_settings :main, :bool => { :type => :boolean, :default => true, :desc => "desc" }
     end
 
+    it "the cache should be threadsafe" do
+      original_cache = @settings.instance_variable_get(:@cache).value
+      other_request = Thread.new do
+        expect(@settings.instance_variable_get(:@cache).value).not_to be(original_cache)
+      end
+      other_request.join
+      expect(@settings.instance_variable_get(:@cache).value).to be(original_cache)
+    end
+
+    it "the cache can be updated with content to be shared with new threads" do
+      @settings[:myval]
+      @settings.unsafe_snapshot_cache
+      original_cache = @settings.instance_variable_get(:@cache).value
+      other_request = Thread.new do
+        expect(@settings.instance_variable_get(:@cache).value).to eq(original_cache)
+        expect(@settings.instance_variable_get(:@cache).value).not_to be(original_cache)
```

Comment:
I'm not really understanding what is expected to change over the course of these two `expect`s.. I know we're doing things in a new thread, but I'm not seeing what would change this or anything to prevent races..?

Reply:
There is a difference between `eq` and `be`:

> Object identity
> expect(actual).to be(expected) # passes if actual.equal?(expected)

> Object equivalence
> expect(actual).to eq(expected) # passes if actual == expected

Reply:
oooh, gotcha. I can never keep straight what the different ones mean, but that makes sense. thanks!